File: py/sort/insert_sort.py
from sort import *


def insert_sort(arr):
    i = 1
    while i < len(arr):
        if arr[i] < arr[i - 1]:
            tmp = arr[i]
            j = i - 1
            while j >= 0 and tmp < arr[j]:
                arr[j + 1] = arr[j]
                j = self_incr(j, -1)

            arr[j + 1] = tmp

        i = self_incr(i)


if __name__ == '__main__':
    arr = [12, 3, 5, 13, 12232, 57, 899, 22, 1, 32]

    insert_sort(arr)

    print(arr)

File: go/sort/insert_sort.go
package main

import "fmt"

// insert sort
func main() {

	var arr = [] int{12, 34, 5, 122, 4, 1, 56, 7, 3, 78, 1}
	insert_sort(arr)
	fmt.Println(arr)
}

func insert_sort(arr []int) {

	var i, j int

	for i = 1; i < len(arr); i++ {

		if arr[i-1] > arr[i] {

			var tmp = arr[i]

			for j = i - 1; j >= 0 && arr[j] > tmp; j-- {
				arr[j+1] = arr[j]

			}

			arr[j+1] = tmp

		}
	}
}
